File: backend/app/schemas/artifact.py
import uuid
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class ArtifactResponse(BaseModel):
    id: str
    assessment_id: str
    stage_name: str
    scanner_name: Optional[str] = None
    command: Optional[str] = None
    parameters: Optional[dict] = None
    scanner_version: Optional[str] = None
    target: Optional[str] = None
    start_time: Optional[str] = None
    end_time: Optional[str] = None
    duration: Optional[float] = None
    status: str
    error_message: Optional[str] = None
    artifact_path: str
    output_type: Optional[str] = None
    hash: Optional[str] = None
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class ArtifactListResponse(BaseModel):
    data: list[ArtifactResponse]
    pagination: dict


class ArtifactContentResponse(BaseModel):
    id: str
    stage_name: str
    content_type: str
    content: Any
    filename: str

File: frontend/src/types/artifact.ts
export interface Artifact {
  id: string;
  assessment_id: string;
  stage_name: string;
  scanner_name: string | null;
  command: string | null;
  parameters: Record<string, unknown> | null;
  scanner_version: string | null;
  target: string | null;
  start_time: string | null;
  end_time: string | null;
  duration: number | null;
  status: string;
  error_message: string | null;
  artifact_path: string;
  output_type: string | null;
  hash: string | null;
  created_at: string;
  updated_at: string;
}

export interface ArtifactListResponse {
  data: Artifact[];
  pagination: {
    page: number;
    per_page: number;
    total: number;
    total_pages: number;
  };
}

export interface ArtifactFile {
  filename: string;
  size: number;
  modified_at: string;
}

export interface ArtifactContent {
  id: string;
  stage_name: string;
  content_type: string;
  content: string;
  filename: string;
}
